src/main/webapp/search.js

```diff
@@ -43,9 +43,54 @@ async function getSearchResultsHelper(document, window) {
     }
 
     if(topic != null) {
-        await fetch("/search?topic="+topic).then(response => response.text()).then((results) => {
+        await fetch("/search?topic="+topic).then(response => response.json()).then((results) => {
             var resultContainer = document.getElementById("result-container");
-            resultContainer.innerHTML = results;
+
+            var numSearchResults = document.createElement("h4");
+            numSearchResults.id = "num-search-results";
+
+            resultContainer.appendChild(numSearchResults);
+
+            //if there was an error reported by the servlet, display the error message
+            if(results.error) {
```

Comment:
I didn't know about this error method. Looks helpful! I'll implement that in my js function too then :)

Reply:
I just return a JSON string with an error attribute in my servlet if the user's search was invalid. I don't believe it has that attribute automatically.

Reply:
Oh I see it now! Good idea